DPPP/test/unit/tBDABuffer.cc

```diff
@@ -0,0 +1,391 @@
+    const bool fr_flags[] {false};
+    const std::complex<float> data {1};
+    const float weights[] {1};
+
+    buffer.AddRow(0., 1., 0, 0, n_channels, n_correlations, &data, flags, weights, fr_flags);
+
+    BOOST_CHECK_EQUAL(*buffer.GetData(size_t {0}), data);
+    BOOST_CHECK_EQUAL(*buffer.GetFlags(size_t {0}), flags[0]);
+    BOOST_CHECK_EQUAL(*buffer.GetFullResFlags(size_t {0}), fr_flags[0]);
+    BOOST_CHECK_EQUAL(*buffer.GetWeights(size_t {0}), weights[0]);
+}
+
+BOOST_AUTO_TEST_CASE( test_modify_data ) 
+{
+    const size_t n_channels {2};
+    const size_t n_correlations {4};
+    BDABuffer buffer {n_channels * n_correlations};
+    const std::complex<float> data {1};
+
+    buffer.AddRow(0., 1., 0, 0, n_channels, n_correlations, &data);
+
+    const std::complex<float> new_data {2};
+    *buffer.GetData() = new_data;
+    BOOST_CHECK_EQUAL(*buffer.GetData(), new_data);
+}
+
+BOOST_AUTO_TEST_CASE( test_modify_flags )
+{
+    const size_t n_channels {2};
+    const size_t n_correlations {4};
+    BDABuffer buffer {n_channels * n_correlations};
+    bool flags[] {false, true, false};
+
+    buffer.AddRow(0., 1., 0, 0, n_channels, n_correlations, nullptr, flags);
+    
+    bool new_flags[] {true, false, true};
+    *buffer.GetFlags() = new_flags[0];
+    *(buffer.GetFlags() + 1) = new_flags[1];
+    *(buffer.GetFlags() + 2) = new_flags[2];
+    BOOST_CHECK_EQUAL(*buffer.GetFlags(), new_flags[0]);
+    BOOST_CHECK_EQUAL(*(buffer.GetFlags() + 1), new_flags[1]);
+    BOOST_CHECK_EQUAL(*(buffer.GetFlags() + 2), new_flags[2]);
+}
+
+BOOST_AUTO_TEST_CASE( test_modify_weights ) 
+{
+    const size_t n_channels {2};
+    const size_t n_correlations {4};
+    BDABuffer buffer {n_channels * n_correlations};
+    float weights[] {1, 2, 3};
+
+    buffer.AddRow(0., 1., 0, 0, n_channels, n_correlations, nullptr, nullptr, weights);
+
+    float new_weights[] {4, 5, 6};
+    *buffer.GetWeights() = new_weights[0];
+    *(buffer.GetWeights() + 1) = new_weights[1];
+    *(buffer.GetWeights() + 2) = new_weights[2];
+    BOOST_CHECK_EQUAL(*buffer.GetWeights(), new_weights[0]);
+    BOOST_CHECK_EQUAL(*(buffer.GetWeights() + 1), new_weights[1]);
+    BOOST_CHECK_EQUAL(*(buffer.GetWeights() + 2), new_weights[2]);
+}
+
+BOOST_AUTO_TEST_CASE( test_modify_full_resflags )
+{
+    const size_t n_channels {2};
+    const size_t n_correlations {4};
+    BDABuffer buffer {n_channels * n_correlations};
+    bool flags[] {false, true, false};
+
+    buffer.AddRow(0., 1., 0, 0, n_channels, n_correlations, nullptr, nullptr, nullptr, flags);
+    
+    bool new_flags[] {true, false, true};
+    *buffer.GetFullResFlags() = new_flags[0];
+    *(buffer.GetFullResFlags() + 1) = new_flags[1];
+    *(buffer.GetFullResFlags() + 2) = new_flags[2];
+    BOOST_CHECK_EQUAL(*buffer.GetFullResFlags(), new_flags[0]);
+    BOOST_CHECK_EQUAL(*(buffer.GetFullResFlags() + 1), new_flags[1]);
+    BOOST_CHECK_EQUAL(*(buffer.GetFullResFlags() + 2), new_flags[2]);
+}
```

Comment:
These tests mainly test that GetXXX()  returns the same pointer when called multiple times. I wouldn't go that far with testing, so please remove these tests.